quickwit/quickwit-proto/build.rs

```diff
@@ -49,25 +45,71 @@ fn main() -> Result<(), Box<dyn std::error::Error>> {
         )
         .type_attribute("OutputFormat", "#[serde(rename_all = \"snake_case\")]")
         .out_dir("src/")
-        .compile_with_config(
-            prost_config,
-            &[
-                "./proto/ingest_api.proto",
-                "./proto/metastore_api.proto",
-                "./proto/search_api.proto",
-            ],
-            &["./proto"],
-        )?;
-    tonic_build::configure().out_dir("src/").compile(
-        &[
-            "./otlp/opentelemetry/proto/common/v1/common.proto", // Must be compiled first.
-            "./otlp/opentelemetry/proto/resource/v1/resource.proto", // Must be compiled second.
-            "./otlp/opentelemetry/proto/logs/v1/logs.proto",
-            "./otlp/opentelemetry/proto/trace/v1/trace.proto",
-            "./otlp/opentelemetry/proto/collector/logs/v1/logs_service.proto",
-            "./otlp/opentelemetry/proto/collector/trace/v1/trace_service.proto",
-        ],
-        &["./otlp"],
-    )?;
+        .compile_with_config(prost_config, &protos, &["./proto"])?;
+
+    // Jaeger proto
+    let protos = ["./jaeger/model.proto", "./jaeger/storage.proto"];
+    let includes = find_jaeger_includes();
+
+    let mut prost_config = prost_build::Config::default();
+    prost_config.type_attribute("Operation", "#[derive(Eq, Ord, PartialOrd)]");
+
+    tonic_build::configure()
+        .out_dir("src/")
+        .compile_with_config(prost_config, &protos, &includes)?;
+
+    // OTEL proto
+    let protos = [
+        "otlp/opentelemetry/proto/common/v1/common.proto", // Must be compiled first.
+        "otlp/opentelemetry/proto/resource/v1/resource.proto", // Must be compiled second.
+        "otlp/opentelemetry/proto/logs/v1/logs.proto",
+        "otlp/opentelemetry/proto/trace/v1/trace.proto",
+        "otlp/opentelemetry/proto/collector/logs/v1/logs_service.proto",
+        "otlp/opentelemetry/proto/collector/trace/v1/trace_service.proto",
+    ];
+    tonic_build::configure()
+        .type_attribute(".", "#[derive(Serialize, Deserialize)]")
+        .out_dir("src/")
+        .compile(&protos, &["./otlp"])?;
     Ok(())
 }
+
+fn find_homebrew_protoc_include_dir() -> Option<PathBuf> {
+    if let Some(protoc_version) = fs::read_dir("/opt/homebrew/Cellar/protobuf")
+        .ok()?
+        .flatten()
+        .map(|dir_entry| dir_entry.file_name())
+        .sorted()
+        .last()
+    {
+        let protoc_include_dir = Path::new("/opt/homebrew/Cellar/protobuf")
+            .join(protoc_version)
+            .join("include");
+        println!(
+            "Homebrew protoc include dir: {}",
+            protoc_include_dir.display()
+        );
+        Some(protoc_include_dir)
+    } else {
+        None
+    }
+}
+
+fn find_jaeger_includes() -> Vec<PathBuf> {
+    let mut includes = vec![PathBuf::from("./jaeger")];
+
+    if let Ok(protoc_include_dir) = env::var("QW_PROTOC_INCLUDE_DIR") {
+        includes.push(PathBuf::from(protoc_include_dir));
```

Comment:
That little dance is for google well-known types right?
I suggest we remove all this logic and just copy the google protos in the repo. 
They are not more sacred than the jaeger protos imho, and our build process will be less brittle
and less environment dependent.